cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -135,7 +142,13 @@ public synchronized void create(NamespaceMeta metadata) throws Exception {
     }
 
     try {
-      storageProviderNamespaceAdmin.create(metadata);
+      impersonator.doAs(namespace, new Callable<Object>() {
```

Comment:
I made UGIProvider operate on the user-level now (instead of namespace level).
However, Impersonator is the class that wraps lookup of this user as well as getting the UGI for this user, so it'll operate on the namespace level.